rust-src/wallet_library/src/credential.rs

```diff
@@ -111,6 +117,102 @@ fn build_policy(
     })
 }
 
+/// The details for a new credential deployment.
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CredentialDeploymentDetails {
+    expiry:       TransactionTime,
+    unsigned_cdi: UnsignedCredentialDeploymentInfo<IpPairing, ArCurve, AttributeKind>,
+}
+
+/// Defines the byte serialization of the credential deployment details. Note
+/// that this serialization is for a new credential being deployed, which the
+/// explicit 0 byte denotes.
+impl Serial for CredentialDeploymentDetails {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        self.unsigned_cdi.serial(out);
+        0u8.serial(out);
+        self.expiry.seconds.serial(out);
+    }
+}
+
+/// Serializes the credential deployment details. It is this serialization that
+/// should be signed when creating a new credential for deployment. The
+/// serialized bytes are returned hex encoded.
+pub fn serialize_credential_deployment_details(
```

Comment:
I'm wondering whether we need this specifically, or just the hash of this which I think is the thing that is actually signed.

Reply:
We just need the hash. I am refactoring it.